src/sql/magic.py

```diff
@@ -406,22 +408,58 @@ def interactive_execute_wrapper(**kwargs):
         # only expand CTE if this is not a CTE itself
         if not is_cte:
             if args.with_:
+                query_type = (
+                    sqlparse.parse(command.sql)[0].get_type()
+                    if sqlparse.parse(command.sql)
+                    else None
+                )
                 with_ = args.with_
             else:
                 with_ = self._store.infer_dependencies(command.sql_original, args.save)
                 if with_:
                     command.set_sql_with(with_)
+                    query_type = (
+                        sqlparse.parse(command.sql_original)[0].get_type()
+                        if sqlparse.parse(command.sql_original)
+                        else None
+                    )
+
+                    if query_type != "SELECT":
+                        raise exceptions.UsageError(
+                            f"Cannot use snippets with {query_type}"
+                        )
+
                     display.message(
                         f"Generating CTE with stored snippets: {pretty_print(with_)}"
                     )
                 else:
+                    query_type = (
+                        sqlparse.parse(command.sql)[0].get_type()
+                        if sqlparse.parse(command.sql)
+                        else None
+                    )
                     with_ = None
         else:
+            query_type = (
+                sqlparse.parse(command.sql)[0].get_type()
+                if sqlparse.parse(command.sql)
+                else None
+            )
             if args.with_:
                 raise exceptions.UsageError(
                     "Cannot use --with with CTEs, remove --with and re-run the cell"
                 )
 
+            """
```

Comment:
This code is wrapped in a string (I guess you want to delete it?), please double check your PRs before requesting a review.

Reply:
It is logic related to snippet expansion that I was not sure if I should implement yet. I'll remove it.